src/main/scala/tech/cryptonomic/conseil/util/DatabaseUtil.scala

```diff
@@ -2,64 +2,154 @@ package tech.cryptonomic.conseil.util
-    */
-  def insertValuesIntoSqlAction[T](xs: Seq[T]): SQLActionBuilder = {
-    var b = sql"("
-    var first = true
-    xs.foreach { x =>
-      if(first) first = false
-      else b = concatenateSqlActions(b, List(sql","))
-      b = concatenateSqlActions(b, List(sql"'#$x'"))
+    /** Creates SQLAction of sequence of values
+      *
+      * @param  xs list of values to be inserted into SQLAction
+      * @return SqlActionBuilder with values from parameter
+      */
+    def insertValuesIntoSqlAction[T](xs: Seq[T]): SQLActionBuilder = {
+      var b = sql"("
+      var first = true
+      xs.foreach { x =>
+        if (first) first = false
+        else b = concatenateSqlActions(b, List(sql","))
+        b = concatenateSqlActions(b, List(sql"'#$x'"))
+      }
+      concatenateSqlActions(b, List(sql")"))
     }
-    concatenateSqlActions(b, List(sql")"))
-  }
 
-  /** Implicit value that allows getting table row as Map[String, Any] */
-  implicit val getMap: GetResult[Map[String, Any]] = GetResult[Map[String, Any]](positionedResult => {
-    val metadata = positionedResult.rs.getMetaData
-    (1 to positionedResult.numColumns).flatMap(i => {
-      val columnName = metadata.getColumnName(i).toLowerCase
-      val columnValue = positionedResult.nextObjectOption
-      columnValue.map(columnName -> _)
-    }).toMap
-  })
+    /** Implicit value that allows getting table row as Map[String, Any] */
+    implicit val getMap: GetResult[Map[String, Any]] = GetResult[Map[String, Any]](positionedResult => {
+      val metadata = positionedResult.rs.getMetaData
+      (1 to positionedResult.numColumns).flatMap(i => {
+        val columnName = metadata.getColumnName(i).toLowerCase
+        val columnValue = positionedResult.nextObjectOption
+        columnValue.map(columnName -> _)
+      }).toMap
+    })
 
-  /** Implicit class providing helper methods for SQLActionBuilder */
-  implicit class SqlActionHelper(action: SQLActionBuilder) {
-    /** Method for adding predicates to existing SQLAction
-      * @param predicates list of predicates to add
-      * @return           new SQLActionBuilder containing given predicates
+    /** Implicit class providing helper methods for SQLActionBuilder */
+    implicit class SqlActionHelper(action: SQLActionBuilder) {
+      /** Method for adding predicates to existing SQLAction
+        *
+        * @param predicates list of predicates to add
+        * @return new SQLActionBuilder containing given predicates
+        */
+      def addPredicates(predicates: List[Predicate]): SQLActionBuilder = {
+        concatenateSqlActions(action, makePredicates(predicates))
+      }
+
+      /** Method for adding ordering to existing SQLAction
+        *
+        * @param ordering list of QueryOrdering to add
+        * @return new SQLActionBuilder containing ordering statements
+        */
+      def addOrdering(ordering: List[QueryOrdering]): SQLActionBuilder = {
+        val queryOrdering = if (ordering.isEmpty) {
```

Comment:
I suggest moving this check within `makeOrdering` as per so-called [postel's law](https://devopedia.org/postel-s-law).

The gist of it, in this case, is to avoid forcing the caller to verify general constraints as pre-conditions to the arguments, while having the target method handle corner cases if it can be made gracefully

Reply:
ok, I've changed it, but I'm not 100% convinced, because this way I'm creating empty `SQLActionBuilder` and merging it when there is no ordering defined

Reply:
You're right, I didn't realize that `makeOrdering` was returning a single action, instead of a list...

Reply:
reverted